ZohoCRMCharts/Classes/ZCRMCommonUtil.swift

```diff
@@ -0,0 +1,226 @@
+//
+//  ZCRMCommonUtil.swift
+//  ZohoCRMCharts
+//
+//  Created by Sarath Kumar Rajendran on 24/10/18.
+//  Copyright © 2018 Zoho CRM. All rights reserved.
+//
+import UIKit
+
+/**
+	Error.
+*/
+public struct ZCRMChartsError : Error {
+	
+	private let message: String
+	
+	init(message: String) {
+		self.message = "ZCRMCharts - \(message)"
+	}
+	
+	public var localizedDescription: String {
+		return self.message
+	}
+}
+
+internal protocol KPIUtil {
+	
+	var type: ZCRMCharts.ZCRMKPIType { get set}
+}
+
+internal extension KPIUtil {
+	
+	var isScorecard: Bool {
+		
+		if self.type == .scorecard {
+			return true
+		}
+		return false
+	}
+	
+	var isRankings: Bool {
+		
+		if self.type == .rankings {
+			return true
+		}
+		return false
+	}
+	
+	var isBasic: Bool {
+		
+		if self.type == .basic {
+			return true
+		}
+		return false
+	}
+	
+	var isStandard: Bool {
+		
+		if self.type == .standard {
+			return true
+		}
+		return false
+	}
+	
+	var isGrowthIndex: Bool {
+		
+		if self.type == .growthIndex {
+			return true
+		}
+		return false
+	}
+}
+
+internal protocol ZCRMLayoutConstraintDelegate: class {
+	
+	var viewConstraints:[NSLayoutConstraint] { get set }
+	
+	func activate(constraints: [NSLayoutConstraint], _ append: Bool)
+	
+	func deactivateConstraints()
+}
+
+internal extension ZCRMLayoutConstraintDelegate {
+	
+	func activate(constraints: [NSLayoutConstraint], _ append: Bool = false) {
+		if append {
+			self.viewConstraints += constraints
+		} else  {
+			self.viewConstraints = constraints
+		}
+		NSLayoutConstraint.activate(constraints)
+	}
+	
+	func deactivateConstraints() {
+		if !self.viewConstraints.isEmpty {
+			NSLayoutConstraint.deactivate(self.viewConstraints)
+		}
+	}
+}
+
+internal extension UIView {
+	
+	internal struct tag {
```

Comment:
add comments.

Reply:
Will do once all the work has been done.